src/platform/plugins/shared/discover/public/application/main/data_fetching/fetch_all.ts

```diff
@@ -89,6 +91,20 @@ export function fetchAll(
     const isEsqlQuery = isOfAggregateQueryType(query);
     const currentTab = getCurrentTab();
 
+    let fetchQuery = query!;
+
+    const internalStateStore = internalState.getState();
+    const currentTabState = internalStateStore.tabs.byId[internalStateStore.tabs.unsafeCurrentId];
+
+    // if the query is an ESQL query and there are cascade groupings selected,
+    // we mutate the query to only include the selected groupings
+    if (isEsqlQuery && currentTabState.uiState.cascadedDocuments?.selectedCascadeGroups?.length) {
+      fetchQuery = mutateQueryStatsGrouping(
+        query,
+        currentTabState.uiState.cascadedDocuments.selectedCascadeGroups
+      );
+    }
```

Comment:
When the user has cascade grouping selected, we modify the query that gets fired, we do this so that the query in the editor doesn't need to change, ensuring that the user is presented still with the query they'd inputed.